pkg/controller/servicemesh/webhooks/validation/controlplane.go

```diff
@@ -146,13 +146,13 @@ func (v *ControlPlaneValidator) validateUpdate(ctx context.Context, old, new *ma
 	// fails because feature X is no longer supported, but was added back in 1.3).
 	if oldVersion.Version() < newVersion.Version() {
 		for version := oldVersion.Version(); version < newVersion.Version(); version++ {
-			if err := v.validateUpgrade(ctx, version, old); err != nil {
+			if err := v.validateUpgrade(ctx, version, new); err != nil {
```

Comment:
I'm not sure I understand this part. If I'm reading the comment right, shouldn't we be validating an upgrade from `version -> version+1`? For that, we'd need to set `new.Spec.Version = version+1` in every loop iteration

Reply:
Since this code is in the 1.1 branch, the for loop is misleading, since there are only two possible versions. The body of the loop is only executed once.

Also, the validation code never looks at `.spec.version`, so there's no need to update it here.

What this PR fixes is the scenario where the user does two things in the same update operation:
- they change the version, and
- make the required changes to the SMCP so that its spec is valid for the _new_ version. 

Previously, users would have to make the changes to the SMCP first and only then change the version.

Reply:
Ah, I get it. Thanks @luksa 

I guess we'll need to implement a `validateUpgrade(fromVersion, toVersion, smcp)` func for maistra-1.2 then

Reply:
This has changed with the version refactoring in maistra-1.2.  Each version strategy will validate downgrade from it to the previous version and upgrade from the previous version to it, e.g. 1.2 strategy would validate upgrade from 1.1 and downgrade to 1.1.  The looping logic persists in 1.2, but as @luksa said, with only two versions, it only loops through once.